operators/src/processing/point_in_polygon.rs

```diff
@@ -115,28 +119,69 @@ impl PointInPolygonFilterProcessor {
+
+        let mut result = vec![false; points.len()];
+
+        thread_pool.scope(|scope| {
+            let num_features = points.len();
+            let feature_offsets = points.feature_offsets();
+            let time_intervals = points.time_intervals();
+            let coordinates = points.coordinates();
+
+            for (chunk_index, chunk_result) in (&mut result).chunks_mut(chunk_size).enumerate() {
+                let feature_index_start = chunk_index * chunk_size;
+                let features_index_end = min(feature_index_start + chunk_size, num_features);
+                let tester = tester.clone();
+
+                scope.compute(move || {
+                    for (
+                        feature_index,
+                        ((coordinates_start_index, coordinates_end_index), time_interval),
+                    ) in two_tuple_windows(
+                        feature_offsets[feature_index_start..=features_index_end]
+                            .iter()
+                            .map(|&c| c as usize),
+                    )
+                    .zip(time_intervals[feature_index_start..features_index_end].iter())
+                    .enumerate()
+                    {
+                        let is_multi_point_in_polygon_collection = coordinates
+                            [coordinates_start_index..coordinates_end_index]
+                            .iter()
+                            .any(|coordinate| {
+                                tester.is_coordinate_in_any_polygon(coordinate, time_interval)
+                            });
 
-        let tester = PointInPolygonTester::new(polygons);
+                        chunk_result[feature_index] = is_multi_point_in_polygon_collection;
+                    }
+                });
+            }
+        });
 
-        let coordinates = points.coordinates();
+        result
+    }
 
-        for ((coordinates_start_index, coordinates_end_index), time_interval) in
-            two_tuple_windows(points.feature_offsets().iter().map(|&c| c as usize))
-                .zip(points.time_intervals())
-        {
-            let is_multi_point_in_polygon_collection = coordinates
-                [coordinates_start_index..coordinates_end_index]
-                .iter()
-                .any(|coordinate| tester.is_coordinate_in_any_polygon(coordinate, time_interval));
+    async fn filter_points(
+        ctx: &dyn QueryContext,
+        points: Arc<MultiPointCollection>,
+        polygons: MultiPolygonCollection,
+        initial_filter: &BooleanArray,
+    ) -> Result<BooleanArray> {
+        let thread_pool = ctx.thread_pool();
 
-            filter.push(is_multi_point_in_polygon_collection);
-        }
+        let thread_points = points.clone();
+        let filter = tokio::task::spawn_blocking(move || {
+            let tester = Arc::new(PointInPolygonTester::new(&polygons)); // TODO: multithread
+            Self::filter_parallel(&tester, &thread_points, &thread_pool)
```

Comment:
You could just move the tester in the method.

Reply:
b4087ec0a91cee8d5f4773e37be84c6839cd51cc